source/logging/logging.rst

```diff
@@ -0,0 +1,380 @@
+change needed.
+
+Drivers SHOULD take advantage of built-in support for logging configuration within logging
+frameworks or their language ecosystem if available. If unavailable, drivers MUST support
+configuration via the listed fallback implementation methods.
+
+In addition to supporting configuration without code changes/recompilation, drivers MAY
+additionally provide support for enabling and configuring logging via their API in a
+manner that does require code changes, and MAY support additional configuration options beyond
+those defined in this specification.
+
+For languages/ecosystems where libraries depend on only a logging interface (for example, the Java
+driver and `SLF4J <https://www.slf4j.org/>`_) and application developers must choose a log handler,
+the requirements MAY be considered satisfied if there is a log handler available in the ecosystem
+which users can select that satisfies the requirements.
+
+Per-Component Configuration of Log Levels
+-----------------------------------------
+Drivers MUST support enabling logging and specifying the minimum severity level for emitted messages
+on a per-component level, for each component defined below in the `Components`_ section.
+  
+  **Fallback implementation method**: Support configuration via environment variables 
+  corresponding to each component, as defined in `Components`_, as well as via the
+  environment variable ``MONGODB_LOG_ALL``.
+
+  Each of these variables may be set to any of the values defined below in `Log Severity Levels`_
+  to indicate the minimum severity level at which messages should be emitted for the
+  corresponding component, or in the case of ``MONGODB_LOG_ALL``, all components. Setting a value
+  for ``MONGODB_LOG_ALL`` is equivalent to setting that value for all of the per-component
+  variables.
+
+  If ``MONGODB_LOG_ALL`` is specified in addition to one or more component variables, the
+  component variable(s) MUST take precedence.
+
+  The default is to not log anything. 
+
+  If a variable is set to an invalid value, it MUST be treated as if it were not specified at all.
+
+Configurable Log Destination
+----------------------------
+Drivers MUST support configuring where log messages should be output, including the options: stdout,
+stderr, output file (path MUST be configurable).
+
+    **Fallback implementation method**: If the environment variable ``MONGODB_LOG_PATH`` is provided:
+     -  If the value is "stdout" (case-insensitive), log to stdout.
+     - If the value is "stderr" (case-insensitive), log to stderr.
+     - Else, log to a file at the specified path. If the file already exists, it MUST be appended to.
+
+    If the variable is not provided, the driver MUST log to stderr.
+
+Configurable Max Document Length
+--------------------------------
+Drivers MUST support configuring the maximum logged length for extended JSON documents in log messages.
```

Comment:
Consider specifying somewhere that Extended JSON Relaxed mode MUST be used for representing BSON documents.